compiler/x/codegen/OpBinary.cpp

```diff
@@ -911,3 +911,904 @@ const TR_X86OpCode::TR_OpCodeBinaryEntry TR_X86OpCode::_binaryEncodings[IA32NumO
       {0x00, 0x00, 0x00, 0},    // AdjustFramePtr
       {0x00, 0x00, 0x00, 0}     // ReturnMarker
    };
+
+#include "codegen/FrontEnd.hpp"
+bool TR_X86OpCode::_UseDeprecatedBinaryEncoding = feGetEnv("TR_UseDeprecatedBinaryEncoding");
+// X87 is only available on X86-32
+#ifdef TR_TARGET_64BIT
+#define X87_________________(x) 0,0,0
+#else
+#define X87_________________(x) (uint8_t)((x & 0xE0) >> 5), (uint8_t)((x & 0x18) >> 3), (uint8_t)(x & 0x07)
+#endif
+
+const TR_X86OpCode::OpCode_t TR_X86OpCode::_binaries[] =
+{
+    { VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_____, 0xcc, 0, ModRM_NONE, Immediate_0 },    // BADIA32Op/int 3
```

Comment:
I completely agree that relying on the order to correlate binaries to its meaning is extremely dangerous. We have this problem not only in the binary table but also in op properties, op names, etc. I have been thinking about this for awhile and am eager to get this restructured so that we do not rely on the order.
However, I believe this should be handled in a separate issue or pull request. I'm concerned that it may be more confusing and problematic if we resolve the binary table but not other places. Another reason why the initial version of new binary table is still rely on order is to be consist with the old design. In this way, we can keep the old approach so that we can verify the new implementation with the old one.
The plan to fully move away  from depending on orders is on the to-do list but I have got a chance fully think about it yet. One idea is to use macro expansion tricks like what we use for Optimizations.enum.